### PR TITLE
deobfuscate: rename accompanying (smaller) files with same basename, and no renaming of collections with same extension

### PR DESCRIPTION
**Pull request by sanderjo**
Closes https://github.com/sabnzbd/sabnzbd/issues/1821

### CODE REVIEW

File: sabnzbd/deobfuscate_filenames.py [hunk 1]
```diff
@@ -158,7 +161,12 @@ def deobfuscate_list(filelist, usefulname):
     # No par2 files? Then we try to rename qualifying (big, not-excluded, obfuscated) files to the job-name
     if run_renamer:
         logging.debug("Trying to see if there are qualifying files to be deobfuscated")
+        # sort filelist on size, so start with biggest file
+        filelist = sorted(filelist, key=os.path.getsize, reverse=True)
```
Comment (Safihre) [inline sabnzbd/deobfuscate_filenames.py:165]: Maybe we can remove the `sorted` a few lines above?

File: sabnzbd/deobfuscate_filenames.py [hunk 2]
```diff
@@ -168,11 +176,19 @@ def deobfuscate_list(filelist, usefulname):
...
-                # OK, rename
+                # Rename and make sure the new filename is unique
                 path, file = os.path.split(filename)
                 new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), get_ext(filename)))
                 logging.info("Deobfuscate renaming %s to %s", filename, new_name)
-                # Rename and make sure the new filename is unique
                 renamer(filename, new_name)
+                # find other files with the same basename (in the same directory), and rename them in the same way:
+                basedirfile, _ = os.path.splitext(filename)
+                for samebasenamefile in glob.glob(basedirfile + "*"):
```
Comment (Safihre) [inline sabnzbd/deobfuscate_filenames.py:186]: Please use `filesystem.globber()`.

**Comment by Safihre** [comment]
If you could add some unittests, that would be great. This kind of added functionality is easily broken.

**Comment by sanderjo** [comment]
Closes #1821

**Comment by neofright** [comment]
Unfortunately @sanderjo the linuxserver.io 'latest' tag is still lagging behind master.

I'll reply back when the container is updated to include these commits (if the issue is still open!) but I can't say how long that would be.

**Pull request merged**